libs/functions/generate_api.py

```diff
@@ -83,75 +85,68 @@ def _generate_api_for_projections(projection_row):
             "currentUsage": None,
-            "typicalUsageRate": state_beds_data[BedsDataset.Fields.ALL_BED_TYPICAL_OCCUPANCY_RATE],
-        },
-        ICUBeds={
+            "typicalUsageRate": actual_data.get(CommonFields.ALL_BED_TYPICAL_OCCUPANCY_RATE),
+        }
+    icu_beds = None
+    if CommonFields.ICU_BEDS in actual_data and _get_or_none(actual_data[CommonFields.ICU_BEDS]): 
+        icu_beds = {
             # Note(Chris): We do not currently pass through ICU Bed capacity calculations
             # in the projection_row.  This wouldn't be a ton of work to do, but
             # using the provided beds data for the time being.
-            "capacity": state_beds_data[BedsDataset.Fields.ICU_BEDS],
-            "currentUsage": None,
-            "typicalUsageRate": state_beds_data[BedsDataset.Fields.ICU_TYPICAL_OCCUPANCY_RATE],
-        },
-    )
-
-
-def _generate_county_actuals(projection_row, state_intervention, county_beds_data):
-    intervention_str = state_intervention.name
-
-
-    icu_beds = None
-    if county_beds_data:
-        icu_beds = {
-            "capacity": county_beds_data[BedsDataset.Fields.ICU_BEDS],
+            "capacity": actual_data[CommonFields.ICU_BEDS],
             "currentUsage": None,
-            "typicalUsageRate": county_beds_data[BedsDataset.Fields.ICU_TYPICAL_OCCUPANCY_RATE],
+            "typicalUsageRate": actual_data.get(CommonFields.ICU_TYPICAL_OCCUPANCY_RATE),
         }
 
     return _Actuals(
-        population=projection_row[rc.POPULATION],
+        population= actual_data.get(CommonFields.POPULATION),
         intervention=intervention_str,
-        cumulativeConfirmedCases=projection_row[rc.CURRENT_CONFIRMED],
-        cumulativeDeaths=projection_row[rc.CURRENT_DEATHS],
-        cumulativePositiveTests=None,
-        cumulativeNegativeTests=None,
-        hospitalBeds={
-            "capacity": projection_row[rc.PEAK_BED_CAPACITY],
-            "currentUsage": None,
-            "typicalUsageRate": county_beds_data.get(BedsDataset.Fields.ALL_BED_TYPICAL_OCCUPANCY_RATE),
-        },
+        cumulativeConfirmedCases=_get_or_none(actual_data[CommonFields.CASES]),
+        cumulativeDeaths=_get_or_none(actual_data[CommonFields.DEATHS]),
+        cumulativePositiveTests=_get_or_none(actual_data.get(CommonFields.POSITIVE_TESTS)),
+        cumulativeNegativeTests=_get_or_none(actual_data.get(CommonFields.NEGATIVE_TESTS)),
+        hospitalBeds=hospital_beds,
         ICUBeds=icu_beds,
     )
 
+def _generate_actuals_timeseries( actuals_timeseries_dataset, intervention): 
+    actual_timeseries_api_response = []
+    for row in actuals_timeseries_dataset: 
+        actual = _generate_actuals(row, intervention.name)
+        timeseries_actual = CANActualsTimeseriesRow(**actual.dict(), date=row[CommonFields.DATE])
+        actual_timeseries_api_response.append(timeseries_actual)
+    return actual_timeseries_api_response
+
+def _generate_state_actuals(
```

Comment:
I think if you were to pass in `latest_actuals` here and do the logic for choosing which source to get the data for in the calling functions, you could just call `_generate_actuals` everywhere instead of needing another county/state split